### PR TITLE
updated types for VerifyOptions, Add `onlyCookie` to `VerifyOptions`

### PR DESCRIPTION
**Pull request by mage1k99**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)
      
      
#### Credits

I have added the unit tests that are missing in #260

### CODE REVIEW

File: jwt.d.ts [hunk 1]
```diff
@@ -140,7 +141,7 @@ export interface FastifyJwtDecodeOptions {
 
 declare module 'fastify' {
   interface FastifyInstance {
-    jwt: JWT
+    jwt: JWT & { [namespace: string]: JWT }
```
Comment (mcollina) [inline jwt.d.ts:144]: what does this mean in practice?

**Review by mcollina — CHANGES_REQUESTED** [review]
Can you add a test for the types? We use tsd.

File: jwt.d.ts (hunk 1 above)
Comment (mage1k99) [reply jwt.d.ts]: It does not serve a good purpose, so removing it. I also check with typescript documentation. it was mistake, removing it

Comment (mage1k99) [reply jwt.d.ts]: I have updated, can you check it ?

**Comment by ZiiMakc** [comment]
@mage1k99 Hi, if you don't plan to finish MR I can take it over.

**Comment by mage1k99** [comment]
@ZiiMakc Yeah take over, I'll close this pr once you've done

File: jwt.d.ts [hunk 2]
```diff
@@ -65,6 +65,7 @@ export interface SignOptions extends Omit<SignerOptions, "expiresIn" | "notBefor
 
 export interface VerifyOptions extends Omit<VerifierOptions, "maxAge"> {
   maxAge: number | string;
+  onlyCookie: boolean;
```
Comment (mcollina) [inline jwt.d.ts:68]: This needs to be tested with `tsd`.

**Review by mcollina — CHANGES_REQUESTED** [review]

**Pull request closed without merging**